engine-types/src/types/gas.rs

```diff
@@ -87,6 +87,7 @@ impl Div<u64> for EthGas {
     type Output = Self;
 
     fn div(self, rhs: u64) -> Self::Output {
+        assert!(rhs != 0, "ZERO_IS_AN_INVALID_DENOMINATOR");
         Self(self.0 / rhs)
```

Comment:
I would replace it with [`u64::checked_div`](https://doc.rust-lang.org/std/primitive.u64.html#method.checked_div) method.
```rust
self.0.checked_div(res).map(Self).expect("ZERO_IS_AN_INVALID_DENOMINATOR")
```